components/brave_wallet/browser/zcash/rust/BUILD.gn

```diff
@@ -57,6 +70,17 @@ source_set("orchard_impl") {
   ]
 }
 
+source_set("shard_store") {
+  visibility = [ ":*" ]
+  sources = [ "cxx/src/shard_store.h" ]
```

Comment:
we don't want to use cxx directories anymore and it's not clear to me why this would go somewhere other than the existing `ochrard_impl` target. In a separate PR we should move all the files in `orchard_impl` into an `internal` directory 

Also generally speaking you shouldn't create a target in one directory that includes files in a subdirectory, especially if those are the only files it includes.

Reply:
>  it's not clear to me why this would go somewhere other than the existing ochrard_impl target. 

There will be cycle dependency otherwise rust_lib depends on shard_store.h and orchard_impl depends on rust_lib

Reply:
I would use `qr_code_generator` as an example to follow for this which would probably mean shard_store.h moves into this directory.

Reply:
or actually I think it would move into the internal directory because it's code that directly accesses cxx types, but I'm confused because there appear to be a bunch of cpp files under the `rust` directory so where are we putting the cpp wrappers? We can mix rs and cpp in the same directory, but cpp code should not be in a directory called `rust`

Reply:
it's a bit confusing that this directory is called `rust` and also contains cpp files, but we should probably rename it in a separate PR

Reply:
Moved to the zcash/rust directory